repository/src/main/resources/alfresco/subsystems/Search/common-search-context.xml

```diff
@@ -64,8 +64,8 @@
         </property>
         <!-- Query collections should be loaded on demand using this component - once loaded thay are available for use -->
     </bean>
-    
-    <bean id="search.fts.alfresco.switching" class="org.alfresco.repo.search.impl.solr.DbOrIndexSwitchingQueryLanguage" >
+
+    <bean id="base.search.fts.alfresco.switching" abstract="true" >
```

Comment:
In this place class type definition `class="org.alfresco.repo.search.impl.solr.DbOrIndexSwitchingQueryLanguage"` can stay either. This way in overridden beans it won't be necessary to define it.